File: GO/core/model/votesboxquestion/votesboxquestion.model.go
package votesboxquestionmodel

import (
	"justasking/GO/core/model/votesboxquestionanswer"
	"time"

	uuid "github.com/satori/go.uuid"
)

// VotesBoxQuestion is a votesboxquestion
type VotesBoxQuestion struct {
	QuestionId uuid.UUID
	BoxId      uuid.UUID
	Header     string
	IsActive   bool
	SortOrder  int
	CreatedAt  time.Time
	CreatedBy  string
	UpdatedAt  *time.Time
	UpdatedBy  string
	DeletedAt  *time.Time
	Answers    []votesboxquestionanswermodel.VotesBoxQuestionAnswer `gorm:"-"`
}

// TableName returns the table name for use with ORM
func (VotesBoxQuestion) TableName() string {
	return "votes_box_questions"
}

File: web/src/app/core/models/votes-box-question.model.ts
import { VotesBoxQuestionAnswerModel } from './votes-box-question-answer.model'

export class VotesBoxQuestionModel {
    Answers?: VotesBoxQuestionAnswerModel[];
    BoxId?: string;
    Header?: string;
    IsActive?: boolean; 
    QuestionId?: string;
    SortOrder?: number; 
    SelectedAnswerId?: string;
}
